DisCatSharp/Clients/Rest/BaseMethods/Invites/DiscordApiClient.BaseMethods.Invites.cs

```diff
@@ -0,0 +1,186 @@
+// This file is part of the DisCatSharp project, based off DSharpPlus.
+//
+// Copyright (c) 2021-2022 AITSYS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading.Tasks;
+
+using DisCatSharp.Entities;
+using DisCatSharp.Net.Abstractions;
+using DisCatSharp.Net.Serialization;
+
+using Newtonsoft.Json;
+
+namespace DisCatSharp.Net;
+
+public sealed partial class DiscordApiClient
+{
+	/// <summary>
+	/// Gets the guild invites async.
+	/// </summary>
+	/// <param name="guildId">The guild_id.</param>
+
+	internal async Task<IReadOnlyList<DiscordInvite>> GetGuildInvitesAsync(ulong guildId)
+	{
+		var route = $"{Endpoints.GUILDS}/:guild_id{Endpoints.INVITES}";
+		var bucket = this.Rest.GetBucket(RestRequestMethod.GET, route, new {guild_id = guildId }, out var path);
+
+		var url = Utilities.GetApiUriFor(path, this.Discord.Configuration);
+		var res = await this.ExecuteRestRequest(this.Discord, bucket, url, RestRequestMethod.GET, route).ConfigureAwait(false);
+
+		var invitesRaw = JsonConvert.DeserializeObject<IEnumerable<DiscordInvite>>(res.Response).Select(xi => { xi.Discord = this.Discord; return xi; });
+
+		return new ReadOnlyCollection<DiscordInvite>(new List<DiscordInvite>(invitesRaw));
+	}
+
+	/// <summary>
+	/// Gets the invite async.
+	/// </summary>
+	/// <param name="inviteCode">The invite_code.</param>
+	/// <param name="withCounts">If true, with_counts.</param>
+	/// <param name="withExpiration">If true, with_expiration.</param>
+	/// <param name="guildScheduledEventId">The scheduled event id to get.</param>
+
+	internal async Task<DiscordInvite> GetInviteAsync(string inviteCode, bool? withCounts, bool? withExpiration, ulong? guildScheduledEventId)
```

Comment:
So this could be a runner up for converting into a fluent-api approach / request record

3 out of 4 parameters here look to be "optional" so if I wanted just an invite code (didn't care about the rest)
```csharp
GetInviteAsync("meowmeow", null, null, null);
```
Looks janky?